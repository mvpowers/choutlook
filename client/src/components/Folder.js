import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-direction: row;
  padding: 5px;
  align-items: center;
  color: #666;
  font-weight: 400;
  font-size: 14px;

  &:hover {
    background-color: #eaeaea;
    color: #000;
  }
`;

const Icon = styled.div`
  padding: 15px;
`;

const Name = styled.div`
  display: flex;
  align-items: center;
  padding-right: 15px;
`;

const Folder = ({ name, icon }) => (
  <Container>
    <Icon>{icon}</Icon>
    <Name>{name}</Name>
  </Container>
);

Folder.propTypes = {
  name: PropTypes.string.isRequired,
  icon: PropTypes.shape({
    props: PropTypes.shape({
      size: PropTypes.number,
    }),
  }).isRequired,
};

export default Folder;
