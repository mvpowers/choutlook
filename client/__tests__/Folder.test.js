import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Folder from '../src/components/Folder';

Enzyme.configure({ adapter: new Adapter() });

describe('Folder', () => {
  it('should be defined', () => {
    expect(Folder).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(<Folder name="test" icon={{ props: { size: 16 } }} />);
    expect(tree).toMatchSnapshot();
  });
});
