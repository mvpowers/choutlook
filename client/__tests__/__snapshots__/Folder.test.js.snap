// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Folder
    icon={
      Object {
        "props": Object {
          "size": 16,
        },
      }
    }
    name="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Folder__Icon>
          Object {
            "props": Object {
              "size": 16,
            },
          }
        </Folder__Icon>,
        <Folder__Name>
          test
        </Folder__Name>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Object {
            "props": Object {
              "size": 16,
            },
          },
        },
        "ref": null,
        "rendered": Object {
          "props": Object {
            "size": 16,
          },
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Folder__Icon>
            Object {
              "props": Object {
                "size": 16,
              },
            }
          </Folder__Icon>,
          <Folder__Name>
            test
          </Folder__Name>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Object {
              "props": Object {
                "size": 16,
              },
            },
          },
          "ref": null,
          "rendered": Object {
            "props": Object {
              "size": 16,
            },
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
